### ISSUE TITLE
when http URL is missing a slash in the report

### ISSUE THREAD

**Issue by karlcow** [webcompat/webcompat.com#767]
This is minor. It happened once in 
https://github.com/webcompat/web-bugs/issues/1807

Or logic for parsing the URL and displaying the domain in the title of an issue fails when the URL is having an incorrect http. I'm merely filling the issue to have  a record of the problem. It good be a good first patch. 

To make it clearer.

The person typed/reported:

```
https:/amazon.com
```

And then our code parsed it as 

```
https: 
```

creating a title which is `https: - layout is messed up` instead of `amazon.com - layout is messed up`

**Comment by hallvors** [comment]
I suggest a simple JS replace in the client-side form validation code - in the URL field change handler.

**Comment by deckycoss** [comment]
i'll try working on this patch...

**Comment by miketaylr** [comment]
The code that parses a URL into a domain lives here: https://github.com/webcompat/webcompat.com/blob/master/webcompat/form.py#L127

**Comment by miketaylr** [comment]
(I would start with that, I think that's the right place -- @karlcow can confirm ^_^)

**Comment by karlcow** [comment]
And this one will fail too.
https://github.com/webcompat/webcompat.com/blob/master/webcompat/form.py#L118

I suggest to create tests first for these, then patch. :)

**Comment by karlcow** [comment]
* `input` -> `normalize` -> `domain`
* `example.com` -> `http://example.com/` -> `example.com`
* `http:/example.com` -> `http://example.com/` -> `example.com`
* `https:/example.com` -> `https://example.com/` -> `example.com`
* `http:example.com` -> `http://example.com/` -> `example.com`
* `https:example.com` -> `http://example.com/` -> `example.com`

**Comment by deckycoss** [comment]
two questions:

1. is it fair to update the "SCHEMES" tuple so that it includes the misspelled protocols, or should i create a new tuple for it?
2. @karlcow, in your examples, the output of normalize_url always contains a trailing slash. should i add this functionality to normalize_url? (i'm assuming that you would only want to add trailing slashes if there is no path after the hostname.)

**Comment by miketaylr** [comment]
(I'll let @karlcow answer these questions -- he's more knowledgeable about the representing URIs and URLs and HTTP, etc. Just be a little patient @deckycoss, Karl is based in Tokyo so it's his weekend right now).

**Comment by karlcow** [comment]
1. About Schemes tuple  
It's better to keep things separated, aka what is valid from what we are helping to fix. So I would recommend to keep the SCHEMES tuple as is. 
2. About trailing slashes  
To the best of my knowledge, many servers will try to add a trailing slash as a first step with a redirection. It's not mandatory though. But I had time to time some servers who were not answering without a trailing slash. On the other hand we might want to go along with what the user says. Let's not take care of the trailing slash and if we need to fix that we will do it later. I fixed the tests in the comment.
3. About https/http in the edit.  
Just fixed the test for this. Thanks for catching that.
4. the `//`  
Added another example. This is a valid way of making a link. it's useful when a server serves both https and http. And fortunately python knows how to deal with it.

```python
>>> url = '//example.com/foo/bar'
>>> import urlparse
>>> urlparse.urlparse(url).netloc
'example.com'
```
